sdk/trace/tracer.go

```diff
@@ -127,7 +128,8 @@ func (tr *tracer) newRecordingSpan(psc, sc trace.SpanContext, name string, sr Sa
 		spanKind:               trace.ValidateSpanKind(config.SpanKind()),
 		name:                   name,
 		startTime:              startTime,
-		attributes:             newAttributesMap(tr.provider.spanLimits.AttributeCountLimit),
+		attributes:             make(map[attribute.Key]attribute.Value),
+		maxAttributes:          tr.provider.spanLimits.AttributeCountLimit,
```

Comment:
I see here that the span stores `tr`. Similar to my first question, can we eliminate `maxAttributes`, `resource`, and `spanLimits` fields? These are call copied from the tracer, but we keep a reference to the tracer so we may not need these copies.

Reply:
I'm happy to include changes to how `maxAttributes` is represented in this PR, but will hold off changes to `resource` and `spanLimits` for another PR dedicated to these changes.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/2575